### PR TITLE
adding some more typing

### PR DESCRIPTION
**Pull request by bbarker**
WIP to improve typing

Test additions forthcoming

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "sqlalchemy-stubs"]
+	path = sqlalchemy-stubs
+	url = https://github.com/JelleZijlstra/sqlalchemy-stubs
```
Comment (erickpeirson) [inline .gitmodules]: Let's move this to requirements, per discussion.

File: default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#
+# This file is just a script to setup the environment using Nix and is not typically used
+# in practice; a more well known alternative for environment setup and packaging can be found
+# in the Dockerfile. For more information see https://nixos.org/nix/ and
+# https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.md
+#
```
Comment (erickpeirson) [inline default.nix]: Per our discussion, let's keep this out of master/develop for now -- orphan branch is fine.

**Comment by bbarker** [comment]
@erickpeirson the review comments have been addressed - thanks!

**Comment by erickpeirson** [comment]
Merging!

**Pull request merged**